airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_core.py

```diff
@@ -520,6 +520,24 @@ def test_oauth_flow_parameters(self, actual_connector_spec: ConnectorSpecificati
 
         diff = paths_to_validate - set(get_expected_schema_structure(spec_schema))
         assert diff == set(), f"Specified oauth fields are missed from spec schema: {diff}"
+        """
+        OAuth is default check.
+        If credentials do have oneOf: we check that the OAuth is listed at first.
+        If there is no oneOf and Oauth: OAuth is only option to authenticate the source and no check is needed.
+        """
+        if advanced_auth.predicate_key:
+            credentials = advanced_auth.predicate_key[0]
+            one_of_default_method = None
+            try:
+                one_of_default_method = dpath.util.get(spec_schema, f"/**/{credentials}/oneOf/0")
+            except KeyError:  # Key Error when oneOf is not in credentials object
+                pass
+            if one_of_default_method:
+                path_in_credentials = "/".join(advanced_auth.predicate_key[1:])
+                auth_method_predicate_const = dpath.util.get(one_of_default_method, f"/**/{path_in_credentials}/const")
+                assert (
```

Comment:
I think this assertion will fail for source-zendesk-talk (and maybe others) as [the first element](https://github.com/airbytehq/airbyte/blob/670ffdd32c5869fed3fd8c6d6d9787de84104965/airbyte-integrations/connectors/source-zendesk-talk/source_zendesk_talk/spec.json#L21-L43) is not oauth. How will we get visibility on sources that will fail because of this?

Reply:
thanks for pointing this out. i will add possibility to pass skip and bypass reason for this test  in case there is a reason that OAuth is not default method. Health engineer will be responsible to fix sources that fail due to this check.